src/test/groovy/org/gradle/android/RoomSchemaLocationKspWorkaroundTest.groovy

```diff
@@ -0,0 +1,203 @@
+package org.gradle.android
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Assume
+import spock.lang.Unroll
+
+import static org.gradle.testkit.runner.TaskOutcome.FROM_CACHE
+import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
+
+@MultiVersionTest
+class RoomSchemaLocationKspWorkaroundTest extends RoomWorkaroundAbstractTest {
+
+    @Unroll
+    def "schemas are generated with Ksp into task-specific directory and are cacheable with kotlin and kapt workers enabled (Android #androidVersion) (Kotlin #kotlinVersion)"() {
+        def kotlinVersionNumber = VersionNumber.parse(kotlinVersion)
+
+        // Kotlin supported versions doesn't include the Ksp 1.6 version
+        Assume.assumeFalse(kotlinVersionNumber <= VersionNumber.parse("1.7.0"))
```

Comment:
I think it is better to use positive assertions. Is 1.7 supported? I think your previous logic might suggest it isn't
```suggestion
        Assume.assumeTrue(kotlinVersionNumber >= VersionNumber.parse("1.7.0"))
```